src/neptune_scale/api/attribute.py

```diff
@@ -0,0 +1,243 @@
+import functools
+import itertools
+import warnings
+from datetime import datetime
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Collection,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+if TYPE_CHECKING:
+    from neptune_scale.api.run import Run
+
```

Comment:
maybe define _all__ = []? 
There are some methods like accumulate_dict_values that seem like something that shouldn't really be public